webproject/src/main/java/data/Room.java

```diff
@@ -0,0 +1,139 @@
+package data;
+
+import java.io.File;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+
+/**
+ * This class defines room objects
+ */
+public class Room {
+  private static Gson gson = new Gson();
+
+  private String id;
+  private String title;
+  private String link;
+  private String description;
+  private int deliveryLocation; //modify to use different representation of location if needed
+  private int phoneNumber; //modify to use different representation of contact details if needed
+  private String category;
+  private String imagePath;
+  private RoomType roomType;
+  private RoomCondition roomCondition;
+
+  private Room (Builder builder) {
```

Comment:
it is better to have a function newBuilder() to create the builder, it is also useful when you want to apply default value
```
private Builder newBuilder() {
  return new Builder().setSomeParameter(defaultValue);
}
```

this way, the code will be something like
```
Room.newBuilder().setSomeParameter(...).setSomeParameter(...).build()
```

Reply:
Ok! should it be public to create the object in other classes?